### PR TITLE
feat: add support for overrideConfig in programmatic API

### PR DESCRIPTION
**Pull request by adidahiya**
I used eslint-interactive recently to programmatically suppress lint rules across a large codebase. It worked well, but I found it to be a bit slow running all the enabled rules. Sometimes I want to use eslint-interactive with just a few rules enabled, and I found no way to do this with the programmatic API.

This PR adds two new properties accepted in the options object for the `Core()` constructor, `overrideConfig` and `useEslintrc`. These forward directly to the `ESLint()` constructor.

I've also adjusted the code and types a bit in core.ts to make it clear which parts of the API are forwarding to ESLint directly. Please let me know what you think, thanks.

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -79,14 +80,12 @@ export class Core {
...
   constructor(config: Config) {
-    this.config = config;
-    this.baseOptions = {
-      cache: this.config.cache ?? DEFAULT_BASE_CONFIG.cache,
-      cacheLocation: this.config.cacheLocation ?? DEFAULT_BASE_CONFIG.cacheLocation,
-      rulePaths: this.config.rulePaths,
-      extensions: this.config.extensions,
-      cwd: this.config.cwd,
+    this.config = {
+      ...DEFAULT_BASE_CONFIG,
```
Comment (adidahiya) [inline src/core.ts:84]: This fixes a bug where `DEFAULT_BASE_CONFIG.quiet` and `DEFAULT_BASE_CONFIG.formatterName` were only being used in the CLI, and not the programmatic API. I think my new code implements the original intention of `DEFAULT_BASE_CONFIG`.

**Comment by adidahiya** [comment]
I enabled Github Actions, you can check out the builds on my branch here: https://github.com/adidahiya/eslint-interactive/actions?query=branch%3Aad%2FoverrideConfig

**Comment by mizdra** [comment]
Thanks for contributing! I'll review this pull request later!

**Comment by adidahiya** [comment]
There's a performance regression but I'm not sure what it's caused by: https://github.com/adidahiya/eslint-interactive/commit/1db1073b39d81b5dced85e2bdcc7b312454b2eda#commitcomment-82817209

File: src/core.ts [hunk 2]
```diff
@@ -50,23 +50,24 @@ async function getUsedRuleIds(targetFilePaths: string[], options: ESLint.Options
 export type Undo = () => Promise<void>;
 
 /** The config of eslint-interactive */
-export type Config = {
+export type Config = Pick<
+  ESLint.Options,
+  'extensions' | 'rulePaths' | 'cache' | 'cacheLocation' | 'useEslintrc' | 'overrideConfig' | 'cwd'
+> & {
```
Comment (mizdra) [inline src/core.ts:56]: Nice refactoring 👍

File: src/core.ts (hunk 1 above)
Comment (mizdra) [reply src/core.ts:84]: As you say, this is a bug. Thanks for fixing it.

**Comment by mizdra** [comment]
The continuous benchmark is not stable and sometimes failed (ref: https://github.com/mizdra/eslint-interactive/issues/202). You can safely ignore them this time! Sorry for worrying you.

I will raise the upper limit of the benchmark threshold at a later date.

**Review by mizdra — APPROVED** [review]
LGTM!

The programmable API was created just for fun, so I am glad to know that there are people using it.

**Comment by mizdra** [comment]
This PR is released by [v10.1.0](https://github.com/mizdra/eslint-interactive/releases/tag/v10.1.0).

**Comment by adidahiya** [comment]
Thanks for the quick review and release @mizdra! I'll try out the release and follow up if I need more adjustments.

**Pull request merged**